### ISSUE TITLE
dhcp dns suffix support

### ISSUE THREAD

**Issue by jonnytest1** [espressif/arduino-esp32#10387]
### Related area

wifi,HTTPClient

### Hardware specification

esp32

### Is your feature request related to a problem?

my router provides a dns suffix for all my local devices via dhcp, these dont work with my current esp32 implementation

### Describe the solution you'd like

dhcp dns suffix is considered for resolving dns queries

### Describe alternatives you've considered

for now ive manually added the suffix to the request , but it should come from dhcp imo

### Additional context

my current version is 3.20014.231204 so if this added since thn this can be ignored

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

**Comment by me-no-dev** [comment]
Please submit an issue report in the ESP-IDF repository, where the whole network stack actually comes from. It has to be implemented there, before we can pick it up. In case they let you know that there has to be some configuration change that we would need to make, please come back and post a new issue where we will track and implement it. thanks!
